x-pack/plugins/fleet/server/collectors/agent_collectors.ts

```diff
@@ -47,3 +49,97 @@ export const getAgentUsage = async (
     updating,
   };
 };
+
+export interface AgentData {
+  agent_versions: string[];
+  agent_checkin_status: {
+    error: number;
+    degraded: number;
+  };
+  agent_checkin_status_last_1h: {
+    error: number;
+    degraded: number;
+  };
+}
+
+const DEFAULT_AGENT_DATA = {
+  agent_versions: [],
+  agent_checkin_status: { error: 0, degraded: 0 },
+  agent_checkin_status_last_1h: { error: 0, degraded: 0 },
+};
+
+export const getAgentData = async (esClient?: ElasticsearchClient): Promise<AgentData> => {
```

Comment:
it will be a great improvements to write an integration tests for these collectors(I think it's doable with a jest integration test). 
Maybe this can be done in a follow up issue if it's too much work.

Reply:
I'll look into adding integration tests.

I think I might create a separate pr for the agent logs collector too, since that depends on ES change.

Reply:
Added integration test. 
I wonder how could we improve the integration test framework, it seems rather slow to start up an ES instance for every test. And it is slow to test changes locally too.